File: src/com/l2kt/gameserver/skills/conditions/ConditionTargetRaceId.java
package com.l2kt.gameserver.skills.conditions;

import java.util.List;

import com.l2kt.gameserver.model.actor.Npc;
import com.l2kt.gameserver.skills.Env;

/**
 * @author nBd
 */
public class ConditionTargetRaceId extends Condition
{
	private final List<Integer> _raceIds;
	
	public ConditionTargetRaceId(List<Integer> raceId)
	{
		_raceIds = raceId;
	}
	
	@Override
	public boolean testImpl(Env env)
	{
		if (!(env.getTarget() instanceof Npc))
			return false;
		
		return _raceIds.contains(((Npc) env.getTarget()).getTemplate().getRace().ordinal());
	}
}

File: src/com/l2kt/gameserver/skills/conditions/ConditionTargetRaceId.kt
package com.l2kt.gameserver.skills.conditions

import com.l2kt.gameserver.model.actor.Npc
import com.l2kt.gameserver.model.actor.template.NpcTemplate
import com.l2kt.gameserver.skills.Env

class ConditionTargetRaceId(private val _raceIds: List<Int>) : Condition() {

    override fun testImpl(env: Env): Boolean {
        return if (env.target !is Npc) false else _raceIds.contains(((env.target as Npc).template as NpcTemplate).race.ordinal)

    }
}